client/src/companiesES/Company/EmailMasks/AddEmailMask.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import { useEntitlementSdk } from '../../../entitlementSdkContext';
+import * as models from '../../../client/es';
+import { Textbox } from '../../../components/Textbox';
+import { Status } from '../../../components/Status/Status';
+import { Error } from '../../../components/Error/Error';
+import styles from '../company.module.scss';
+
+const defaultState: AddEmailMaskState = {
+  emailMask: '',
+  errors: [],
+};
+
+interface OwnProps {
+  companyId: string;
+  updateEmailMasks: (ips: models.EmailMask[]) => void;
+}
+
+export type AddEmailMaskProps = OwnProps;
+
+interface OwnState {
+  emailMask: string;
+  errors: models.ApiErrors;
+}
+
+export type AddEmailMaskState = OwnState;
+
+const AddEmailMask: React.FC<AddEmailMaskProps> = ({
+  companyId,
+  updateEmailMasks,
+}) => {
+  const [state, setState] = React.useState<AddEmailMaskState>(defaultState);
+  const [isMaskValid, setIsMaskValid] = React.useState<boolean | undefined>();
+  const [pending, setPending] = React.useState(false);
+  const [status, setStatus] = React.useState<'success' | 'error'>();
+
+  const entitlementSdk = useEntitlementSdk();
+
+  React.useEffect(() => {
+    setIsMaskValid(undefined);
+    if (state.emailMask.length > 0) {
+      const validationDelay = setTimeout(async () => {
+        const isValid = validateEmailMask(state.emailMask);
+        setIsMaskValid(isValid);
+      }, 1000);
+      return () => clearTimeout(validationDelay);
+    }
+    return;
+  }, [state.emailMask]);
+
+  const onClickAddEmailMask = () => {
```

Comment:
```suggestion
  const onAddEmailMaskClick = () => {
```
Sounds more accurate 🤠